modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java

```diff
@@ -631,17 +631,15 @@ public Result run0(
                 }
 
                 // Retry must reset these fields when we release the whole branch without remove.
-                assert r.needReplaceInner == FALSE : "needReplaceInner";
-                assert r.needMergeEmptyBranch == FALSE : "needMergeEmptyBranch";
+                assert !r.needReplaceInner && r.needMergeEmptyBranch == FALSE
+                        : "needReplaceInner=" + r.needReplaceInner + ", needMergeEmptyBranch=" + r.needMergeEmptyBranch;
```

Comment:
Honestly, old approach looked better. Complex conditions are more difficult to read, you should have replaced it with
```
assert !r.needReplaceInner : "needReplaceInner is true";
assert r.needMergeEmptyBranch == FALSE : "needMergeEmptyBranch=" + r.needMergeEmptyBranch;
```

Please do the same for other changed assertions